File: app/src/main/java/com/drivingschoolrwandaapp/models/entities/ReadingSession.java
package com.drivingschoolrwandaapp.models.entities;
import com.google.gson.annotations.SerializedName;
import java.io.Serializable;
import java.util.List;

public class ReadingSession implements Serializable {
    @SerializedName("id")
    private int id;

    @SerializedName("pdfId")
    private int pdfId;

    @SerializedName("currentPage")
    private int currentPage;

    @SerializedName("totalPages")
    private int totalPages;

    @SerializedName("pdf")
    private PdfFile pdf;

    @SerializedName("createdAt")
    private String createdAt;

    @SerializedName("updatedAt")
    private String updatedAt;

    // Getters and Setters
    public int getId() { return id; }
    public int getPdfId() { return pdfId; }
    public int getCurrentPage() { return currentPage; }
    public void setCurrentPage(int currentPage) { this.currentPage = currentPage; }
    public int getTotalPages() { return totalPages; }
    public PdfFile getPdf() { return pdf; }
    public String getCreatedAt() { return createdAt; }
    public String getUpdatedAt() { return updatedAt; }

    public int getProgressPercentage() {
        if (totalPages == 0) return 0;
        return (int) ((currentPage / (float) totalPages) * 100);
    }
}

File: app/src/main/java/com/drivingschoolrwandaapp/models/entities/ReadingSession.kt
package com.drivingschoolrwandaapp.models.entities

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class ReadingSession(
    @SerializedName("id") var id: Int = 0,
    @SerializedName("pdfId") var pdfId: Int = 0,
    @SerializedName("currentPage") var currentPage: Int = 0,
    @SerializedName("totalPages") var totalPages: Int = 0,
    @SerializedName("pdf") var pdf: PdfFile? = null,
    @SerializedName("createdAt") var createdAt: String? = null,
    @SerializedName("updatedAt") var updatedAt: String? = null
) : Serializable {

    fun getProgressPercentage(): Int {
        if (totalPages == 0) return 0
        return ((currentPage.toFloat() / totalPages) * 100).toInt()
    }
}
